### PR TITLE
[automatic failover] Add Dynamic Weight Management for Redis Multi-Database Connections

### PR DESCRIPTION
**Pull request by atakavci**
## Summary
Adds the ability to dynamically update database weights at runtime for multi-database Redis connections, enabling flexible endpoint priority adjustments without reconnection.

### API Enhancements
- Added `setWeight(float)` method to `RedisDatabase` interface
- Made `weight` field mutable in `RedisDatabaseImpl`

### Documentation Updates
- Clarified that weight is used for "endpoint priority" rather than "load balancing" across javadocs
- Updated terminology in `DatabaseConfig`, `BaseRedisMultiDbConnection`, and `RedisDatabase`

### Test Coverage
- Added 6 integration tests for weight management operations
- Added 7 unit tests covering edge cases (zero, negative, max values, multiple updates)
- Tests verify weight retrieval, updates, and multi-database scenarios

## Use Case
Allows runtime adjustment of database priorities for failover scenarios without requiring connection recreation.

### CODE REVIEW

File: src/test/java/io/lettuce/core/failover/StatefulMultiDbConnectionIntegrationTests.java [hunk 1]
```diff
@@ -143,6 +143,111 @@ void shouldAppendToStringSync() {
         connection.close();
     }
 
+    // ============ Database Weight Management Tests ============
```
Comment (ggivo) [inline src/test/java/io/lettuce/core/failover/StatefulMultiDbConnectionIntegrationTests.java:146]: These tests mainly verify basic getter/setter behavior on Database and don’t seem to validate any higher-level selection logic. I’m not sure they add much value as-is.

**Comment by atakavci** [comment]
> When adding or removing databases, we evaluate and switch to the database with the highest weight. However, changing the weights does not currently trigger a re-evaluation or switch. I would expect weight updates to behave similarly to add/remove operations and trigger a switch to the highest-weighted database.

@ggivo adding/removing db does not trigger a failback/failover directly. Right now fundamental parts are there to set/switch/failover in case user likes to do so.

**Comment by ggivo** [comment]
>  adding/removing db does not trigger a failback/failover directly. Right now fundamental parts are there to set/switch/failover in case user likes to do so.

@atakavci 
Agree. 
But this change made me think, if we update the weights of databases, we are missing an API to trigger reevaluation and select the best match.

**Comment by atakavci** [comment]
> But this change made me think, if we update the weights of databases, we are missing an API to trigger reevaluation and select the best match.

@ggivo  I will work on improving with your suggestion, probably next week.
thanks
